test/frontend/components/Fields/BranchField.test.jsx

```diff
@@ -0,0 +1,30 @@
+import { expect } from 'chai';
+
+import { validateBranchName } from '../../../../frontend/components/Fields/BranchField';
+
+describe('<BranchField />', () => {
+  it('validates', () => {
+    const msg = 'Branch name contains invalid characters.';
+
+    expect(validateBranchName()).to.be.undefined;
+    expect(validateBranchName('')).to.be.undefined;
+    expect(validateBranchName('branch')).to.be.undefined;
+    expect(validateBranchName('branch/1')).to.be.undefined;
```

Comment:
while this is a valid git branch name, I think we would want to disallow `/` because it would lead to weird pathing behavior when branches are published to S3 (since those paths look like `.../preview/<repo-owner>/<repo-name>/<branch-name>/...`

Reply:
What is the current behavior if the user adds a repo that has a branch named using that pattern? If the example on line 12 is a valid branch name it's possible a user will add a repo and try to build a branch named something similar right?

I guess my question is: is this actually a problem with the validator

Reply:
I think everything is allowed currently, as long as the site creation process succeeds. We run into problems later, like the inability to create viewable branch preview links (for which https://github.com/18F/federalist/issues/1018 was implemented).

It's not a problem with the validator per se, but could help prevent other later problems if the allowed branch name characters are further restricted.

Reply:
I suppose my ultimate point is that we need to address it more fully in a follow up PR, because in the case described here, we should probably have an additional message letting a user know why we can't build a site using a branch named in this way